components/websubhub/hub_state_update.bal

```diff
@@ -20,73 +20,112 @@ import websubhub.connections as conn;
         common:logError("Error occurred while initializing the hub-state using the latest state-snapshot", httpError, severity = "FATAL");
         return httpError;
     }
 }
 
 function updateHubState() returns error? {
+    log:printDebug("Hub state update worker started", pollingInterval = config:kafka.consumer.pollingInterval);
     while true {
+        log:printDebug("Polling for state update events from Kafka", topic = config:state.events.topic);
         kafka:BytesConsumerRecord[] records = check conn:websubEventsConsumer->poll(config:kafka.consumer.pollingInterval);
         if records.length() <= 0 {
+            log:printDebug("No state update events received in this poll cycle");
             continue;
         }
+        log:printDebug("Received state update events from Kafka", eventCount = records.length());
         foreach kafka:BytesConsumerRecord currentRecord in records {
+            log:printDebug("Processing state update event", offset = currentRecord.offset);
             string lastPersistedData = check string:fromBytes(currentRecord.value);
             error? result = processStateUpdateEvent(lastPersistedData);
             if result is error {
                 common:logError("Error occurred while processing state-update event", result, severity = "FATAL");
                 return result;
+            } else {
+                log:printDebug("State update event processed successfully", offset = currentRecord.offset);
             }
         }
+        log:printDebug("All state update events in batch processed successfully", eventCount = records.length());
     }
 }
 
 function processStateUpdateEvent(string persistedData) returns error? {
+    log:printDebug("Parsing state update event JSON", dataSize = persistedData.length());
     json event = check value:fromJsonString(persistedData);
     string hubMode = check event.hubMode;
-    match event.hubMode {
+    log:printDebug("State update event parsed successfully", hubMode = hubMode);
+    match hubMode {
         "register" => {
+            log:printDebug("Processing topic registration event");
             websubhub:TopicRegistration topicRegistration = check event.fromJsonWithType();
+            log:printDebug("Topic registration event deserialized", topic = topicRegistration.topic);
             check processTopicRegistration(topicRegistration);
+            log:printDebug("Topic registration event processed successfully", topic = topicRegistration.topic);
         }
         "deregister" => {
+            log:printDebug("Processing topic deregistration event");
             websubhub:TopicDeregistration topicDeregistration = check event.fromJsonWithType();
+            log:printDebug("Topic deregistration event deserialized", topic = topicDeregistration.topic);
             check processTopicDeregistration(topicDeregistration);
+            log:printDebug("Topic deregistration event processed successfully", topic = topicDeregistration.topic);
```

Comment:
Can't we have a single log line for this? There is a cost of checking whether debug is enable in each call